### ISSUE TITLE
plot,section-method() should permit plotting distance w.r.t. a 'spine'

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1677]
Consider the attached station-location diagram.  (Ignore the fact that the coastline is not of sufficiently high resolution to put the stations in the water.)

Plotting a section with `xtype="distance"` is a reasonably intuitive way to plot such data, but it is probably not what I would do "by hand".  I would more likely construct some smoothed representation of the ship track -- let me call that the "spine" -- and then plot with respect to distance projected onto that line.

This is not the best illustration, but I'm pretty sure other oceanographers get the idea of what I mean.  (I think oceanviewer has such a thing, actually ... or, anyway, there is some software I've seen that does this.)

My idea is to let the user specify a spine, or perhaps to get oce to compute a spine, and then the "x" axis on the plot will measure distance after moving stations to that line by using a local coordinate system, i.e. a curved coordinate system.

I know I'm not expressing this too clearly.  I'll likely try out some code ideas, since quite often doing that can clarify ideas.

![section](https://user-images.githubusercontent.com/99469/78460761-e7d82f00-7699-11ea-997f-779ec822dcb7.png)

**Comment by richardsc** [comment]
I know what you mean by a "spine", and the "projection" of the stations onto the coordinate system defined by that spine. I think this is a good idea, and it is something I have though about before (but not had a need to do generally ...).

I'm going to tag @clayton33 here, because this seems like either something she already does, or might want to do.

Finally, I see a use case here as it relates to "non-traditional" station data (such as Argo), and how it could be used to create a "virtual section".

(I'm also curious about the section data that you're showing in that map, but guessing that you  could let me know over email rather than in public 😄 )

**Comment by dankelley** [comment]
Yes, good point about argo, which I also had in mind.  This is one of those things that I've never actually *needed*, but that I've wanted quite often.

Initially I was thinking that I'd like using a spline for the spine (gotta type carefully to get this clear!), but in practice I think piecewise linear would be better, because e.g. a person could use a tiny table, or even a sentence, to fully specify the shape, whereas if it's a spline or something, then it would be harder to specify in a reproducible way that would let e.g. a python user do the same thing.

I may play a bit with this today.  Easier to write test code than to explain, and I am not sure yet on the most reliable method (e.g. this seems like an `optimize` problem, but it will not always get the best of several minima).
